### PR TITLE
A pair of fixes related to cursor movement in conpty

### LINKED ISSUE #2642: Emacs cursor displayed incorrectly
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
 Win32NT             10.0.18922.0 Microsoft Windows NT 10.0.18922.0

Windows Terminal version (if applicable):
0.4.2382.0

Any other software?
Ubuntu 18.04.2
GNU Emacs 25.2.2
```

# Steps to reproduce
```
echo "abc
def" > foo
emacs foo
```
Press down arrow, then up arrow.  Cursor is now on 'a' on first line.
Try C-e (control+e).  Cursor should move to end of line but does not appear to move.  Hit right arrow.  Cursor moves to 'd' on 2nd line.

Press up arrow again.  Press right arrow four times.  Cursor remains on first character until 4th press at which point it jumps to 'd' on 2nd line again.

Clearly the cursor is moving but the display is not reflecting that.

Note that as with #283 setting terminal to rxvt or using tmux fixes the issue, however, that bug was marked as fixed in version 1903, which I have.

# Expected behavior

Cursor should move in response to keystrokes.

# Actual behavior

Cursor appears not to move until movement takes it to a new line.

### LINKED ISSUE #4102: Curses programming: cursor handled incorrectly
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.1]
Windows Terminal version : Version: 0.7.3451.0

CPython: 3.7.4
CPython package curses(Windows-curses)
```

# Steps to reproduce
Open `cmd` or `PowerShell` inside Windows Terminal, with `python` run 

```python
def main():
    window = curses.initscr()
    curses.noecho()
    curses.nonl()
    curses.nocbreak()
    curses.raw()
    window.keypad(False)
    x = 0
    y = 0
    while True:
        window.refresh()
        c = window.getch()
        window.addstr(0, 0, str(c))
        if c == ord('q'):
            return
        elif c == ord('a'):
            x = 10
        elif c == ord('b'):
            y = 10
        else:
            x = 0
            y = 0
            pass

        window.move(y, x)
try:
    main()
finally:
    curses.endwin()
```

# Expected behavior

Press `a`, the cursor move to `(0, 10)` immediately.

# Actual behavior

After pressing `a`, the cursor doesn't move to `(0, 10)` immediately, i.e., the screen show doesn't change at all.  Then after an another pressing, the cursor moves to `(0, 10)`. 

P.S:

- Expected behavior is according to the package `curses` bundled in Python of Linux  distros.
- Things work perfectly when we use `cmd` or `PowerShell` without Windows Terminal.

https://github.com/zephyrproject-rtos/windows-curses/issues/7

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

This is a pair of related fixes to conpty. For both of these bugs, the root cause was that the cursor was getting set to Off in conpty. Without the `CursorBlinkerTimer`, the cursor would remain off, and frames that only had cursor movements would not update the cursor position in the terminal.

## PR Checklist
* [x] Closes #4102 
* [x] Closes #2642
* [x] I work here
* [ ] Tests added/passed
* [n/a] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments

Recall that there's a bunch of cursor state that's hard to parse without looking up:
* `Visibility` This controls whether the cursor is visible _at all_, regardless if it's been blinked on or off
* `Blinking` controls whether the blinker timer should do something, or leave the cursor alone.
* `IsOn`: When the cursor is blinking, this alternates between true and false. 

The trick here is that we only `TriggerCursorMoved` when the cursor is `On`, and there are some scenarios where the cursor is manually set to off. 

Fundamentally, these two bugs are similar cases, but they are triggered by different things:
* #2642 was caused by `DoSrvPrivateAllowCursorBlinking(false)` (`^[[?12l`) also manually turning the cursor off.
* #4102 was caused by the client calling `SetConsoleScreenBuffer` to change the active buffer. `win-curses` actually uses that API instead of the alt buffer.

### CODE REVIEW

File: src/host/getset.cpp [hunk 1]
```diff
@@ -1341,7 +1341,6 @@ void DoSrvPrivateShowCursor(SCREEN_INFORMATION& screenInfo, const bool show) noe
 void DoSrvPrivateAllowCursorBlinking(SCREEN_INFORMATION& screenInfo, const bool fEnable)
 {
     screenInfo.GetActiveBuffer().GetTextBuffer().GetCursor().SetBlinkingAllowed(fEnable);
-    screenInfo.GetActiveBuffer().GetTextBuffer().GetCursor().SetIsOn(!fEnable);
```
Comment (DHowett-MSFT) [inline src/host/getset.cpp]: oh interesting, so the old code actually fully disabled the cursor if blinking was disabeld.

i want to make sure, though: _if the user turns blinking off while the cursor is visible, will it get stuck on? if it's off, will it get stuck off?_

Comment (zadjii-msft) [reply src/host/getset.cpp]: From what I've gathered from other terminals, when blinking is disabled, the cursor should remain `On` always, and have the visibility controlled by the `IsVisible` property. So when you do a `printf "\e[?12l"` to disable blinking, the cursor stays stuck `On`. At this point, only the cursor visibility property controls whether the user can see it or not. (Yes, the cursor can be `On` and NOT `Visible`)

Comment (DHowett-MSFT) [reply src/host/getset.cpp]: I don’t understand why it gets stuck _on_ though. What if it’s off (blinked-off) when the blink timer is arrested?

Comment (zadjii-msft) [reply src/host/getset.cpp]: That's a good question - it does seem to get stuck `On`, but if this isn't triggering it, then something else must be. I'll write a test to make sure. Because it _should_ be on.

**Comment by DHowett-MSFT** [comment]
Ref #2697

**Comment by miniksa** [comment]
> Recall that there's a bunck of cursor state that's hard to parse without looking up:

bunck --> bunch
